test/other-issues/custom-user-data/custom-user-data.ts

```diff
@@ -0,0 +1,176 @@
+import sinon from "sinon";
+import {Connection} from "../../../src/connection/Connection";
+import {createTestingConnections, closeTestingConnections, reloadTestingDatabases } from "../../utils/test-utils";
+import { SimpleConsoleLogger, EntityManager } from "../../../src";
+import { Foo } from "./entity/Foo";
+
+describe("other issues > capture custom user data in logger (e.g. trace id)", () => {
+    let connections: Connection[];
+    let manager: EntityManager;
+    let logQueryStub: sinon.SinonStub;
+    const testFoo: Foo = {
+        id: 1,
+        name: "foo",
+    };
+
+    before(async() => {
+        logQueryStub = sinon.stub(SimpleConsoleLogger.prototype, "logQuery");
+        connections = await createTestingConnections({
+            enabledDrivers: ["postgres"],
+            entities: [Foo],
+            logging: true,
+            createLogger: () => new SimpleConsoleLogger(),
+        });
+    });
+    beforeEach(async () => {
+        await reloadTestingDatabases(connections);
+        await Promise.all(connections.map(connection => connection.getRepository(Foo).insert(testFoo)));
+    });
+    afterEach(async () => {
+        await Promise.all(connections.map(connection => connection.getRepository(Foo).delete({})));
+        logQueryStub.resetHistory();
+    });
+    after(() => {
+        closeTestingConnections(connections);
+        logQueryStub.restore();
+    });
+
+    describe("when attaching user data to request-scoped entity manager", () => {
```

Comment:
sorry - quite a lot of test cases here. Different methods on the repository were creating queryRunners in different ways, so I erred on the side of caution and wrote tests for most of them to avoid any surprises.